File: mcst/Variable.m
classdef Variable < Segment
    properties
        name char
        type char
        default Expression
        comment char
    end
    methods
        function self = Variable(name, type, default, comment)
            self.name = name;
            self.type = type;
            self.default = default;
            self.comment = comment;
        end
    end
end

File: test_m/py/nodes/Variable.py
from mruntime import *
from test_m.py.nodes.Segment import Segment
class Variable(Segment):
    def __init__(self, *nargin): # retval: self
        [name, type, default, comment] = nargin
        nargin = len(nargin)
        self.name = name
        self.type = type
        self.default = default
        self.comment = comment
    @staticmethod
    def empty():
        return []
